### PR TITLE
perf(chat_search): build the ChatSearch (CTRL+K) menu once and live update it 

### PR DESCRIPTION
**Pull request by jrainville**
### What does the PR do

Fixes https://github.com/status-im/status-desktop/issues/18297

The ChatSearch menu (ctrl+k menu) now only  gets built once and then gets updated using signals

This should improve it's feel immensely on big accounts, since before, it was rebuilt each time you pressed the hotkey

Contains two commits. The first one moves the chat search to the serach module. The reaosn being that the main module is already gigantic and it makes more sense in the search module. 
It also improves the `build` function a little bit.

The second commit is what makes it build only once and adds events. So I recommend reviewing per commit.

### Affected areas

Codewise:
- main module
- search module

UX wise:
- CTRL+K menu

### Architecture compliance

- [x] I am familiar with the [application architecture](/docs/architecture.md) and agreed good practices.
My PR is consistent with this document: [QML Architecture Guidelines](/guidelines/QML_ARCHITECTURE_GUIDE.md)

### Screencapture of the functionality

Adding and removing channels:

[add-remove-channels.webm](https://github.com/user-attachments/assets/547a624e-5ff2-4a00-a172-c43a4d683995)

Updating channels and comms:

[updating.webm](https://github.com/user-attachments/assets/5fb48b83-a67c-4995-a222-88e5b7fb2f1a)

Menu is super fast now on a big account:

[fast-menu.webm](https://github.com/user-attachments/assets/ac682e57-590a-4d47-b666-d07541061d1d)

### Impact on end user

Way faster CTRL+K menu

### How to test

- Add, remove and update channels and check the menu

### Risk 

Low. Worst case is some edge case of channel updating doesn't work until a restart.

### CODE REVIEW

File: src/app/modules/main/app_search/view.nim [hunk 1]
```diff
@@ -58,3 +62,16 @@ QtObject:
...
   proc emitAppSearchCompletedSignal*(self: View) =
     self.appSearchCompleted()
+
+  proc chatSearchModel*(self: View): chat_search_model.Model =
+    return self.chatSearchModel
+
+  proc buildChatSearchModel*(self: View) {.slot.} =
+    self.delegate.buildChatSearchModel()
+
+  proc chatSearchModelChanged*(self: View) {.signal.}
```
Comment (caybro) [inline src/app/modules/main/app_search/view.nim]: Doesn't really need a notifier signal; it's a pointer that never changes, the contents does

File: ui/app/AppLayouts/stores/RootStore.qml [hunk 2]
```diff
@@ -202,12 +203,12 @@ QtObject {
 
     // Chat related properties and functions that shall be moved to `ChatRootStore`
     property AppSearchStore appSearchStore: AppSearchStore {
-        appSearchModule: internal.mainModuleInst.appSearchModule
+        appSearchModule: internal.appSearchModuleInst
     }
-    property var chatSearchModel: internal.mainModuleInst.chatSearchModel
+    property var chatSearchModel: internal.appSearchModuleInst.chatSearchModel
```
Comment (caybro) [inline ui/app/AppLayouts/stores/RootStore.qml]:
```suggestion
    readonly property var chatSearchModel: internal.appSearchModuleInst.chatSearchModel
```

**Review by caybro — APPROVED** [review]
Looks good, just 2 minor details

File: ui/app/AppLayouts/stores/RootStore.qml [hunk 3]
```diff
@@ -202,12 +203,12 @@ QtObject {
...
     property AppSearchStore appSearchStore: AppSearchStore {
-        appSearchModule: internal.mainModuleInst.appSearchModule
+        appSearchModule: internal.appSearchModuleInst
     }
-    property var chatSearchModel: internal.mainModuleInst.chatSearchModel
+    readonly property var chatSearchModel: internal.appSearchModuleInst.chatSearchModel
 
-    function rebuildChatSearchModel() {
-        internal.mainModuleInst.rebuildChatSearchModel()
+    function buildChatSearchModel() {
```
Comment (micieslak) [inline ui/app/AppLayouts/stores/RootStore.qml]: Is it necessary to expose that method and force UI to decide when it should be built? If it's build once maybe we could hide it?

Comment (jrainville) [reply ui/app/AppLayouts/stores/RootStore.qml]: I like that it leaves the option to lazy load it (only build it when it's necessary).

However, right now, we are indeed building it right on app start because of the Home page needing it. So if we prefer, I can just move the build to the Init and remove all those functions.

File: ui/app/mainui/AppMain.qml [hunk 4]
```diff
@@ -2382,8 +2378,7 @@ Item {
                 searchBoxPlaceholder: qsTr("Where do you want to go?")
                 model: rootStore.chatSearchModel
 
-                onAboutToShow: rootStore.rebuildChatSearchModel()
-                onSelected: function (sectionId, chatId) {
```
Comment (caybro) [inline ui/app/mainui/AppMain.qml:2386]: This line should stay

Comment (caybro) [reply ui/app/mainui/AppMain.qml:2386]: Looks like a wrong rebase :)

Comment (caybro) [reply ui/app/mainui/AppMain.qml:2386]: and I really meant just that one last line:
```qml
onSelected: function (sectionId, chatId) {
```

Comment (jrainville) [reply ui/app/mainui/AppMain.qml:2386]: Sorry about that. I guess it was a rebase issue. Fixed

**Review by micieslak — APPROVED** [review]
nice!
